SimCalorimetry/HcalTrigPrimAlgos/src/HcalTriggerPrimitiveAlgo.cc

```diff
@@ -595,16 +595,17 @@ void HcalTriggerPrimitiveAlgo::analyzeHF2017(
             if (saturated) {
                output[ibin] = QIE10_MAX_LINEARIZATION_ET;
             } else {
-               // If both channels are valid, we cut the sum in half.
-               if (long_fiber_count == 2)
-                  long_fiber_val *= 0.5;
-               if (short_fiber_count == 2)
-                  short_fiber_val *= 0.5;
+               // if one of the dual anode read-outs is valid, double the
+               // energy of the remaining one
+               if (long_fiber_count == 1)
+                  long_fiber_val *= 2;
+               if (short_fiber_count == 1)
+                  short_fiber_val *= 2;
 
                auto sum = long_fiber_val + short_fiber_val;
                // If both towers are valid, we cut the sum in half
-               if (long_fiber_count > 0 and short_fiber_count > 0)
-                  sum *= 0.5;
+               if (long_fiber_count == 0 or short_fiber_count == 0)
```

Comment:
I should update the comment (was focused on logic) to read
```c++
// If one of the fibers is invalid, use the value of the other one instead by doubling the sum
```
Other than that, I can also link to the uHTR spec, which is a bit more verbose and covers a lot more.

Would that be better?